plugins/modules/group.py

```diff
@@ -230,9 +274,10 @@ def create_group(module):
         module.fail_json(**result)
     else:
         msg = "Group: %s SUCCESSFULLY created." % module.params['name']
+        result['changed'] = True
 
     if module.params['group_attributes'] or module.params['user_list_action']:
-        result['msg'] += modify_group(module)
+        msg += modify_group(module)
```

Comment:
why is this if block in here?
I thought if `state` == `modify` then you would then just modify.
and `state` == `present` then it would only create a group.
with this if block, does that mean we can also use `state` == `present`
to modify an existing group.

the intended scenario of when this would happen is:
you run the playbook first time with `state == present` which creates the 
group (you also specified the `group_attributes` and `user_list_action`)
then you run a second time, without changing the `state`, but now that
it exists, it is supposed to try and modify instead.

however, in the main function (line 388) we check if the group exists,
if it does it just reports a message that the group exists, but never actually
goes inside the `create_group`.

or is the intent of this if block, is to set the group attributes and/or user list
after creation of the group?

Reply:
The module doc description explains that you can have group attributes or and actions to the group members either while creating the group or modifying the group.
So yes, the user can create the group with attributes that are modifiable plus it can also add users to the group in the state present, or it can do both things in two different calls, creating the group then modifying the group with the attributes.

Reply:
i see, so this if block, is meant for the case when the user wants to create a group, and then maybe set some group attributes and/or add users to the group all in a single module call.
thanks i understand now.